jujugui/static/gui/src/app/utils/changes-utils.js

```diff
@@ -0,0 +1,242 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2013 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('changes-utils', function(Y) {
+
+  var ns = Y.namespace('juju');
+
+  var removeBrackets = /^\(?(.{0,}?)\)?$/;
+
+  /**
+    Util methods for handling ecs changes.
+
+    @method ChangesUtils
+    @constructor
+  */
+  function ChangesUtils(cfg) {}
+
+  ChangesUtils.prototype = {
+
+    /**
+      Return a list of all change descriptions.
+
+      @method generateAllChangeDescriptions
+      @param {Object} services The list of services from the db.
+      @param {Object} units The list of units from the db.
+      @param {Object} changeSet The current environment change set.
+    */
+    generateAllChangeDescriptions: function(changeSet, services, units) {
+      var changes = [],
+          change;
+      Object.keys(changeSet).forEach(function(key) {
+        change = this.generateChangeDescription(
+            changeSet[key], services, units);
+        if (change) {
+          changes.push(change);
+        }
+      }, this);
+      return changes;
+    },
+
+    /**
+      Return a description of an ecs change for the summary.
+
+      @method generateChangeDescription
+      @param {Object} change The environment change.
+      @param {Object} services The list of services from the db.
+      @param {Object} units The list of units from the db.
+      @param {Bool} skipTime optional, used for testing, don't generate time.
+    */
+    generateChangeDescription: function(change, services, units, skipTime) {
+      var changeItem = {};
+
+      if (change && change.command) {
+        // XXX: The add_unit is just the same as the service because adding
+        // the service also adds the unit. We need to look at the UX for
+        // units as follow up.
```

Comment:
Ah sorry, I didn't even write that, it got copied across when I moved these utils :)